lib/statespace_avx.h

```diff
@@ -111,6 +112,57 @@ struct StateSpaceAVX final : public StateSpace<ParallelFor, float> {
         Base::num_threads_, Base::raw_size_ / 16, f, Op(), state);
   }
 
+  std::vector<uint64_t> Sample(
```

Comment:
Does this make use of any AVX instructions? If not, I think we should provide a shared sampler for all statespace implementations.

Reply:
My concern here isn't different floating-point types (although that's also a valid issue) - it's that statespace_basic has no equivalent sampling method.

Reply:
No, this doesn't make use of any AVX instructions. However, it does make use of specific state vector layout that is relevant only for the AVX version.

Reply:
I will add a sampler to statespace_basic in the next PR.